precompiles/referenda/src/lib.rs

```diff
@@ -253,27 +270,46 @@ where
 		proposal: BoundedBytes<GetCallDataLimit>,
 		block_number: u32,
 	) -> EvmResult {
-		let proposal_origin: GovOrigin = track_id.try_into().map_err(|_| {
-			RevertReason::custom("Origin does not exist for TrackId").in_field("trackId")
-		})?;
-		let proposal_origin: Box<OriginOf<Runtime>> = Box::new(proposal_origin.into());
-		let proposal: BoundedCallOf<Runtime> = Bounded::Inline(
-			frame_support::BoundedVec::try_from(proposal.as_bytes().to_vec()).map_err(|_| {
-				RevertReason::custom("Proposal input is not a runtime call").in_field("proposal")
-			})?,
-		);
-		let enactment_moment = DispatchTime::At(block_number.into());
-
-		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
-
-		let call = ReferendaCall::<Runtime>::submit {
-			proposal_origin,
-			proposal,
-			enactment_moment,
+		{
+			let proposal_origin: GovOrigin = track_id.try_into().map_err(|_| {
+				RevertReason::custom("Origin does not exist for TrackId").in_field("trackId")
+			})?;
+			let proposal_origin: Box<OriginOf<Runtime>> = Box::new(proposal_origin.into());
+			let proposal: BoundedCallOf<Runtime> = Bounded::Inline(
+				frame_support::BoundedVec::try_from(proposal.as_bytes().to_vec()).map_err(
+					|_| {
+						RevertReason::custom("Proposal input is not a runtime call")
+							.in_field("proposal")
+					},
+				)?,
+			);
+			let enactment_moment = DispatchTime::At(block_number.into());
+
+			let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+
+			let call = ReferendaCall::<Runtime>::submit {
+				proposal_origin,
+				proposal,
+				enactment_moment,
+			}
+			.into();
+
+			<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
 		}
-		.into();
 
-		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+		handle.record_log_costs_manual(2, 32 * 2)?;
+		let proposal: sp_std::vec::Vec<u8> = proposal.into();
+		let hash = <Runtime as frame_system::Config>::Hashing::hash(&proposal);
+		log2(
+			handle.context().address,
+			SELECTOR_LOG_SUBMITTED_AT,
+			H256::from_low_u64_be(track_id as u64),
+			EvmDataWriter::new()
+				.write::<u32>(block_number)
+				.write::<H256>(hash.into())
+				.build(),
+		)
+		.record(handle)?;
```

Comment:
what was the reason for emitting the hash of the proposal preimage instead of just the proposal preimage as Bytes32?

The substrate `Submitted` event includes the `proposal` input (without hashing it).

Reply:
Also do we charge for the hashing in the record_log_costs?

Reply:
> reason for emitting the hash

It was described as this in the requirements ticket. I thought about it and makes sense as including a large blob in the log seems unnecessary. Although we can do it if we want to purely align to what frame event looks like.

> do we account for the hashing

We might remove the hash after all but to answer the question, we don't. As we don't _know_ the configured hashing algo, and in case we decide to maintain the hash, do we want to charge some upper bound for it e.g. keccak cost?

Reply:
>  I thought about it and makes sense as including a large blob in the log seems unnecessary.

Sounds reasonable